### DISCUSSION TITLE
How to limit the m3u8 segments downloaded?

### DISCUSSION THREAD

**Discussion by henriqueleite42** [videojs/video.js#7767] [Q&A]
## Description

I want to limit the download of the `.ts` files of a `.m3u8` video, like this:

1 - The user watches a piece of the video
2 - Some `.ts` files are downloaded (like 2 or 3 files)
3 - The download only continues after the user watches some of the `.ts` files
__3.1 - If the user pauses the video, the download stop

So, instead of downloading everything as fast as possible, download the files progressively as the user watches the video.

This is necessary to don't download segments that the user will not watch.

## Example

1 - The user opens the website and the video starts to load
2 - The `.m3u8` file and the playlist are downloaded
3 - The first 3 segments of the video are downloaded
4 - The user watches 2 of the segments
5 - 3 more segments are downloaded
6 - The user watches 1 segment and stops the video
7 - The download stops and no other segment is downloaded
8 - The user starts the video again and watches one more segment
9 - 3 more segments are downloaded

## May-be-related questions but with no solution

- [[HLS] How to temporarily stop buffering #5312](https://github.com/videojs/video.js/issues/5312)

## "Google searches" to (may) help others to find this question in the future

- videojs m3u8 prevent bandwidth waste
- videojs m3u8 only a few segments at time
- videojs m3u8 preload only a few segments
- videojs m3u8 preload only a few ts files
- videojs m3u8 preload ts files
- videojs m3u8 progressive ts files
- videojs m3u8 preload donwload
- videojs m3u8 delay
- videojs m3u8 add delay
- videojs m3u8 only user watches
- videojs delay loading
- videojs hls delay loading
- videojs hls save bandswitch
- videojs hls temporarily stop buffering
- videojs hls stop buffering

**Comment by mustafamunawar** [comment]
Did you find an answer? I though this was the whole purpose of HLS? I inspected DevTools, and I noticed all the segment files were being download. This is actually very costly, so it's very important to solve.  I do know that HLS.js has a "maxBufferLength" that may be of help.
